### PR TITLE
Add web-components and Vue.js to assess

### PR DESCRIPTION
**Pull request by RoryH**

### CODE REVIEW

**Comment by njprrogers** [comment]
👍

File: recommendations/frameworks/JavaScript.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
 # Web
 
 
+##Assess
+
+  - [Web Components](https://www.webcomponents.org/)
+    - [Polymer Project](https://www.polymer-project.org/)
+  - [Vue.js](https://vuejs.org/)
```
Comment (umatrangolo) [inline recommendations/frameworks/JavaScript.md]: What is it? Why you need it? What problem does it solve? Please eli5.

**Comment by RoryH** [comment]
#ELI5

## web-components
The emerging set of standards built in natively into web browsers to allow the creation of reusable UI modules in web applications. Currently only supported by edge browsers. No JS libs required. There are still potentially show-stopping missing features here, like being able to render on the server-side in order to deliver good content inline with SEO expectations.

### Polymer project
Polymer project provides poly-fills for the web-components (above) functionality in browsers. It allow any browsers from IE11+ and iOS Safari 8+ to support web components.

## Vue.js
The newest/hottest JavaScript view rendering engines out there.
* Virtual DOM
* Less JSX-esque code patterns requiring transpilation for runtime.
* Modular, and therefore reusable.

**Comment by umatrangolo** [comment]
👍
